WalletWasabi/Blockchain/TransactionBuilding/TransactionModifierWalletExtensions.cs

```diff
@@ -0,0 +1,322 @@
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using NBitcoin;
+using System.Collections.Generic;
+using System.Linq;
+using WabiSabi.Crypto.Randomness;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.Extensions;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Blockchain.TransactionBuilding;
+
+public static class TransactionModifierWalletExtensions
+{
+	public static BuildTransactionResult CancelTransaction(
+		this Wallet wallet,
+		SmartTransaction transactionToCancel)
+	{
+		var keyManager = wallet.KeyManager;
+		var network = wallet.Network;
+
+		if (!transactionToCancel.IsCancelable(keyManager))
+		{
+			throw new InvalidOperationException("Transaction is not cancelable.");
+		}
+
+		// Take the first own output and if we have it that's where we send all our money to.
+		var ownOutput = transactionToCancel.GetWalletOutputs(keyManager).FirstOrDefault();
+
+		// Calculate the original fee rate and fee.
+		var originalFeeRate = transactionToCancel.Transaction.GetFeeRate(transactionToCancel.GetWalletInputs(keyManager).Select(x => x.Coin).Cast<ICoin>().ToArray());
+		var originalFee = transactionToCancel.Transaction.GetFee(transactionToCancel.WalletInputs.Select(x => x.Coin).ToArray());
```

Comment:
Because different people wrote the different lines :)

https://github.com/zkSNACKs/WalletWasabi/pull/10976/commits/325ad6a98dbd2dedd3938dd0481fec62b506525e

Reply:
done